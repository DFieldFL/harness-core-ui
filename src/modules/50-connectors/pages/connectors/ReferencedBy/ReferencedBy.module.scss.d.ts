/* eslint-disable */
// this is an auto-generated file
declare const styles: {
  readonly table: string
  readonly secretIconCss: string
}
export default styles
