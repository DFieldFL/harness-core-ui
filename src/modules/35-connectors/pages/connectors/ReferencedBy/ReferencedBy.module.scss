.table {
  padding: var(--spacing-small) var(--spacing-xlarge);
  background-color: var(--grey-100);
  .secretIconCss {
    margin-top: 2px !important;
    svg {
      width: 25px;
    }
  }
}
