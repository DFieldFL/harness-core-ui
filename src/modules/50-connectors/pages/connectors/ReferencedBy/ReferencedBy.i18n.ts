export default {
  ENTITY: 'ENTITY',
  ACTIVITY: ' LAST ACTIVITY',
  ReferencedBy: 'Referenced by',
  noData: 'There are no references of this Connector.'
}
