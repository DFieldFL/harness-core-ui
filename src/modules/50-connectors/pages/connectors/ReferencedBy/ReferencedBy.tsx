import React, { useMemo, useState } from 'react'
import ReactTimeago from 'react-timeago'
import { Layout, Text, Icon, Color, IconName } from '@wings-software/uicore'
import type { CellProps, Renderer, Column } from 'react-table'
import Table from '@common/components/Table/Table'

import {
  useListReferredByEntities,
  EntitySetupUsageDTO,
  ResponsePageEntitySetupUsageDTO,
  ListReferredByEntitiesQueryParams
} from 'services/cd-ng'
import { PageSpinner } from '@common/components/Page/PageSpinner'
import { Page } from '@common/exports'
import type { UseGetMockData } from '@common/utils/testUtils'
import { getIconByEntityType, getReferredEntityLabelByType } from '../utils/ConnectorUtils'
import i18n from './ReferencedBy.i18n'
import css from './ReferencedBy.module.scss'

interface ReferencedByProps {
  accountId: string
  entityIdentifier: string | undefined
  entityType: ListReferredByEntitiesQueryParams['entityType']
  mockData?: UseGetMockData<ResponsePageEntitySetupUsageDTO>
}

const RenderColumnEntity: Renderer<CellProps<EntitySetupUsageDTO>> = ({ row }) => {
  const data = row.original
  return (
    <Layout.Vertical>
      <Layout.Horizontal>
        <Icon
          name={getIconByEntityType(data.referredByEntity?.type || '') as IconName}
          size={16}
          className={css.secretIconCss}
        />
        <Text color={Color.BLACK} padding={{ left: 'small' }}>
          {data.referredByEntity?.name}
        </Text>
      </Layout.Horizontal>
      <Text color={Color.GREY_400} font={{ size: 'small' }} padding={{ left: 'xxlarge' }} margin={{ left: 'xsmall' }}>
        ({getReferredEntityLabelByType(data.referredByEntity?.type || '')})
      </Text>
    </Layout.Vertical>
  )
}

const RenderColumnActivity: Renderer<CellProps<EntitySetupUsageDTO>> = ({ row }) => {
  const data = row.original
  return (
    <Layout.Horizontal spacing="small">
      {data.createdAt ? <ReactTimeago date={data.createdAt} /> : null}
    </Layout.Horizontal>
  )
}

const ReferencedBy: React.FC<ReferencedByProps> = props => {
  const [page, setPage] = useState(0)
  const { data, loading, refetch } = useListReferredByEntities({
    queryParams: {
      accountIdentifier: props.accountId,
      identifier: props.entityIdentifier,
      entityType: props.entityType,
      pageIndex: page,
      pageSize: 10
    },
    mock: props.mockData
  })

  const columns: Column<EntitySetupUsageDTO>[] = useMemo(
    () => [
      {
        Header: i18n.ENTITY,
        accessor: 'referredByEntity',
        width: '33%',
        Cell: RenderColumnEntity
      },
      {
        Header: i18n.ACTIVITY,
        accessor: 'createdAt',
        width: '66%',
        Cell: RenderColumnActivity
      }
    ],
    [refetch]
  )
  return (
    <Layout.Vertical height={'calc(100vh - 64px'}>
      {/* Removing for now unless we have search enabled
       <Layout.Horizontal
        padding={{ top: 'large', bottom: 'large', left: 'large', right: 'large' }}
        height={72}
        border={{ bottom: true, color: Color.GREY_300 }}
        flex={{ distribution: 'space-between' }}
      > */}
      {/* <Text font={{ size: 'medium' }} color={Color.DARK_600}>
          {i18n.ReferencedBy}
        </Text> */}

      {/* TODO:  <TextInput
          leftIcon="search"
          placeholder="Search"
            value={searchTerm}
            onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
              setSearchTerm(e.target.value.trim())
            }}
        /> */}
      {/* </Layout.Horizontal> */}
      {!loading ? (
        data?.data?.content?.length ? (
          <Table<EntitySetupUsageDTO>
            className={css.table}
            columns={columns}
            data={data.data.content || []}
            pagination={{
              itemCount: data.data.numberOfElements || 0,
              pageSize: data.data.size || 10,
              pageCount: data.data.totalPages || -1,
              pageIndex: data.data.pageable?.pageNumber || 0,
              gotoPage: pageNumber => setPage(pageNumber)
            }}
          />
        ) : (
          <Page.NoDataCard icon="nav-dashboard" message={i18n.noData} />
        )
      ) : (
        <PageSpinner />
      )}
    </Layout.Vertical>
  )
}
export default ReferencedBy
