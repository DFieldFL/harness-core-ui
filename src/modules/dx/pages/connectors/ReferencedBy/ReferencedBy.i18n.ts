export default {
  ENTITY: 'ENTITY',
  ACTIVITY: ' LAST ACTIVITY',
  ReferencedBy: 'Referenced by'
}
