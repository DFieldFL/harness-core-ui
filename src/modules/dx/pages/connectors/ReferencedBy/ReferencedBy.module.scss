.table {
  padding: var(--spacing-small) var(--spacing-xlarge);
  background-color: var(--grey-100);
  .secretIconCss {
    margin-top: 2px;
    svg {
      width: 25px;
    }
  }
}
