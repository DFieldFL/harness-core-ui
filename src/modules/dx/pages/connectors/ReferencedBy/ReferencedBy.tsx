import React, { useMemo, useState } from 'react'
import ReactTimeago from 'react-timeago'
import { Layout, Text, Icon, Color, IconName } from '@wings-software/uikit'
import type { CellProps, Renderer, Column } from 'react-table'
import Table from 'modules/common/components/Table/Table'

import { useListReferredByEntities, EntityReferenceDTO, ResponsePageEntityReferenceDTO } from 'services/cd-ng'
import { PageSpinner } from 'modules/common/components/Page/PageSpinner'
import { Page } from 'modules/common/exports'
import type { UseGetMockData } from 'modules/common/utils/testUtils'
import { getIconByEntityType, getReferredEntityLabelByType } from '../utils/ConnectorUtils'
import i18n from './ReferencedBy.i18n'
import css from './ReferencedBy.module.scss'

interface ReferencedByProps {
  accountId: string
  entityIdentifier: string | undefined
  mockData?: UseGetMockData<ResponsePageEntityReferenceDTO>
}

const RenderColumnEntity: Renderer<CellProps<EntityReferenceDTO>> = ({ row }) => {
  const data = row.original
  return (
    <Layout.Vertical>
      <Layout.Horizontal>
        <Icon
          name={getIconByEntityType(data.referredByEntityType) as IconName}
          size={16}
          className={css.secretIconCss}
        />
        <Text color={Color.BLACK} padding={{ left: 'small' }}>
          {data.referredByEntityName}
        </Text>
      </Layout.Horizontal>
      <Text color={Color.GREY_400} font={{ size: 'small' }} padding={{ left: 'xxlarge' }} margin={{ left: 'xsmall' }}>
        ({getReferredEntityLabelByType(data.referredByEntityType)})
      </Text>
    </Layout.Vertical>
  )
}

const RenderColumnActivity: Renderer<CellProps<EntityReferenceDTO>> = ({ row }) => {
  const data = row.original
  return (
    <Layout.Horizontal spacing="small">
      {data.createdAt ? <ReactTimeago date={data.createdAt} /> : null}
    </Layout.Horizontal>
  )
}

const ReferencedBy: React.FC<ReferencedByProps> = props => {
  const [page, setPage] = useState(0)
  const { data, loading, refetch } = useListReferredByEntities({
    queryParams: { account: props.accountId, identifier: props.entityIdentifier, page: page, size: 10 },
    mock: props.mockData
  })

  const columns: Column<EntityReferenceDTO>[] = useMemo(
    () => [
      {
        Header: i18n.ENTITY,
        accessor: 'referredEntityName',
        width: '33%',
        Cell: RenderColumnEntity
      },
      {
        Header: i18n.ACTIVITY,
        accessor: 'createdAt',
        width: '66%',
        Cell: RenderColumnActivity
      }
    ],
    [refetch]
  )
  return (
    <Layout.Vertical height={'calc(100vh - 64px'}>
      <Layout.Horizontal
        padding={{ top: 'large', bottom: 'large', left: 'large', right: 'large' }}
        height={72}
        border={{ bottom: true, color: Color.GREY_300 }}
        flex={{ distribution: 'space-between' }}
      >
        <Text font={{ size: 'medium' }} color={Color.DARK_600}>
          {i18n.ReferencedBy}
        </Text>

        {/* TODO:  <TextInput
          leftIcon="search"
          placeholder="Search"
            value={searchTerm}
            onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
              setSearchTerm(e.target.value.trim())
            }}
        /> */}
      </Layout.Horizontal>
      {!loading ? (
        data?.data?.content?.length ? (
          <Table<EntityReferenceDTO>
            className={css.table}
            columns={columns}
            data={data.data.content || []}
            pagination={{
              itemCount: data.data.numberOfElements || 0,
              pageSize: data.data.size || 10,
              pageCount: data.data.totalPages || -1,
              pageIndex: data.data.pageable?.pageNumber || 0,
              gotoPage: pageNumber => setPage(pageNumber)
            }}
          />
        ) : (
          <Page.NoDataCard icon="nav-dashboard" message={i18n.noData} />
        )
      ) : (
        <PageSpinner />
      )}
    </Layout.Vertical>
  )
}
export default ReferencedBy
