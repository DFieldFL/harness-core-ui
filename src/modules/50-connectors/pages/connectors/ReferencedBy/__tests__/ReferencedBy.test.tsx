import React from 'react'
import { render, waitFor, queryByText } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import ReferencedBy from '../ReferencedBy'
import referencedData from './referenced-entities-data.json'

jest.mock('react-timeago', () => () => 'dummy date')

describe('Referenced By', () => {
  test('render for no data', async () => {
    const { container } = render(
      <MemoryRouter>
        <ReferencedBy
          accountId="accountId"
          entityIdentifier="entityIdentifier"
          entityType="Connectors"
          mockData={{
            data: {} as any,
            loading: false
          }}
        />
      </MemoryRouter>
    )
    await waitFor(() => queryByText(container, 'There are no references of this Connector.'))
    expect(container).toMatchSnapshot()
  })
  test('render for data', async () => {
    const { container, getByText } = render(
      <MemoryRouter>
        <ReferencedBy
          accountId="accountId"
          entityIdentifier="entityIdentifier"
          entityType="Connectors"
          mockData={{
            data: referencedData as any,
            loading: false
          }}
        />
      </MemoryRouter>
    )
    expect(getByText('ENTITY')).toBeDefined()
    expect(container).toMatchSnapshot()
  })
})
