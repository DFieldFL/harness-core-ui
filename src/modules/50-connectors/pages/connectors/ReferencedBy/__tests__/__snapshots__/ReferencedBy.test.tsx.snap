// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Referenced By render for data 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main"
  >
    <div
      class="table table"
    >
      <div
        class="header"
        role="row"
      >
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 33%;"
          title="Toggle SortBy"
        >
          ENTITY
          <span
            class="bp3-icon bp3-icon-double-caret-vertical StyledProps--main StyledProps--padding-left-small"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="15"
              viewBox="0 0 16 16"
              width="15"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
        <div
          class="cell"
          colspan="1"
          role="columnheader"
          style="width: 66%;"
          title="Toggle SortBy"
        >
           LAST ACTIVITY
          <span
            class="bp3-icon bp3-icon-double-caret-vertical StyledProps--main StyledProps--padding-left-small"
            icon="double-caret-vertical"
          >
            <svg
              data-icon="double-caret-vertical"
              height="15"
              viewBox="0 0 16 16"
              width="15"
            >
              <desc>
                double-caret-vertical
              </desc>
              <path
                d="M5 7h6a1.003 1.003 0 00.71-1.71l-3-3C8.53 2.11 8.28 2 8 2s-.53.11-.71.29l-3 3A1.003 1.003 0 005 7zm6 2H5a1.003 1.003 0 00-.71 1.71l3 3c.18.18.43.29.71.29s.53-.11.71-.29l3-3A1.003 1.003 0 0011 9z"
                fill-rule="evenodd"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        class="row card"
        role="row"
      >
        <div
          class="cell"
          role="cell"
          style="width: 33%;"
        >
          <div
            class="Layout--vertical StyledProps--main"
          >
            <div
              class="Layout--horizontal StyledProps--main"
            >
              <span
                class="bp3-icon bp3-icon- StyledProps--main secretIconCss"
                icon=""
              >
                <svg
                  data-icon=""
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  
                </svg>
              </span>
              <p
                class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black StyledProps--padding-left-small"
              >
                Sample Secret
              </p>
            </div>
            <p
              class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--font-size-small StyledProps--padding-left-xxlarge StyledProps--margin-left-xsmall"
            >
              (
              
              )
            </p>
          </div>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 66%;"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
          >
            dummy date
          </div>
        </div>
      </div>
      <div
        class="row card"
        role="row"
      >
        <div
          class="cell"
          role="cell"
          style="width: 33%;"
        >
          <div
            class="Layout--vertical StyledProps--main"
          >
            <div
              class="Layout--horizontal StyledProps--main"
            >
              <span
                class="bp3-icon bp3-icon- StyledProps--main secretIconCss"
                icon=""
              >
                <svg
                  data-icon=""
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  
                </svg>
              </span>
              <p
                class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black StyledProps--padding-left-small"
              >
                Sample pipeline
              </p>
            </div>
            <p
              class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--font-size-small StyledProps--padding-left-xxlarge StyledProps--margin-left-xsmall"
            >
              (
              
              )
            </p>
          </div>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 66%;"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
          >
            dummy date
          </div>
        </div>
      </div>
      <div
        class="row card"
        role="row"
      >
        <div
          class="cell"
          role="cell"
          style="width: 33%;"
        >
          <div
            class="Layout--vertical StyledProps--main"
          >
            <div
              class="Layout--horizontal StyledProps--main"
            >
              <span
                class="bp3-icon bp3-icon- StyledProps--main secretIconCss"
                icon=""
              >
                <svg
                  data-icon=""
                  height="16"
                  viewBox="0 0 16 16"
                  width="16"
                >
                  
                </svg>
              </span>
              <p
                class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-black StyledProps--padding-left-small"
              >
                Sample project
              </p>
            </div>
            <p
              class="StyledProps--font StyledProps--main StyledProps--color StyledProps--color-grey400 StyledProps--font-size-small StyledProps--padding-left-xxlarge StyledProps--margin-left-xsmall"
            >
              (
              
              )
            </p>
          </div>
        </div>
        <div
          class="cell"
          role="cell"
          style="width: 66%;"
        >
          <div
            class="Layout--horizontal Layout--layout-spacing-small StyledProps--main"
          >
            dummy date
          </div>
        </div>
      </div>
      <div
        class="Layout--horizontal StyledProps--main Pagination--container StyledProps--padding-top-medium StyledProps--padding-bottom-medium StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-distribution-space-between"
      >
        <span
          class="StyledProps--font StyledProps--main Pagination--text StyledProps--inline"
        >
          Showing 
          <strong>
            10
          </strong>
           of 
          <strong>
            10
          </strong>
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`Referenced By render for no data 1`] = `
<div>
  <div
    class="Layout--vertical StyledProps--main"
  >
    <div
      class="StyledProps--font StyledProps--main StyledProps--flex StyledProps--flex-align-center-center"
      style="width: 100%; height: 100%;"
    >
      <div
        class="Layout--vertical Layout--layout-spacing-medium StyledProps--main"
        style="align-items: center; margin-top: -48px; width: 470px;"
      >
        <span
          class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
        >
          <svg
            height="48"
            viewBox="0 0 27 27"
            width="48"
          >
            <filter
              id="nav-dashboard_svg__a"
            >
              <fecolormatrix
                in="SourceGraphic"
                values="0 0 0 0 0.453512 0 0 0 0 0.490089 0 0 0 0 0.533316 0 0 0 1.000000 0"
              />
            </filter>
            <filter
              id="nav-dashboard_svg__b"
            >
              <fecolormatrix
                in="SourceGraphic"
                values="0 0 0 0 0.453512 0 0 0 0 0.490089 0 0 0 0 0.533316 0 0 0 1.000000 0"
              />
            </filter>
            <g
              fill="none"
              fill-rule="evenodd"
              filter="url(#nav-dashboard_svg__a)"
              transform="translate(-100 -231)"
            >
              <g
                filter="url(#nav-dashboard_svg__b)"
                transform="translate(100 231)"
              >
                <rect
                  height="26"
                  opacity="0.5"
                  rx="5"
                  stroke="#cdcaca"
                  width="26"
                  x="0.5"
                  y="0.5"
                />
                <path
                  d="M13.656 8H19v4h-5.344zm0 12v-6.656H19V20zM7 20v-4h5.344v4zm0-5.344V8h5.344v6.656z"
                  fill="#747d88"
                  fill-rule="nonzero"
                />
              </g>
            </g>
          </svg>
        </span>
        <h2
          class="StyledProps--font StyledProps--font-h2 StyledProps--main StyledProps--font-align-center StyledProps--color StyledProps--color-grey500"
          level="2"
        >
          There are no references of this Connector.
        </h2>
      </div>
    </div>
  </div>
</div>
`;
